Week3/homework/js-exercises/creditCardProject.js

```diff
@@ -0,0 +1,59 @@
+'use strict'
+function validateCreditNumber(num) {
+    const array = [];
+    for(let i = 0; i < num.length; i++){
+        array.push(num[i]);
+    }
+
+    //At least two diffrent numbers should be represented
+    function similarity(array){
+            let firstItem = array[0]
+            let x = array.filter(elements => elements == firstItem).length != array.length ? true : false;
+            return x 
+        }
+
+        //The Sum of all the numbers must be greater than 16:
+        let sum = 0
+        for (let i = 0; i < array.length; i++){
+                let arrayInt = parseInt(array[i])
+                sum += arrayInt
+            }
+
+    switch(false){
+
+        //The length of the number
+        case num.length == 16:
+            console.log(`Invalid! The input ${num} should be 16 characters!`);
+            break;
+
+        //All characters must be numbers
+        case num.match(/^[0-9]+$/) != null:
+            console.log(`Invalid! The input ${num} should contain only numbers!`);
+            break;
+
+        //At least two diffrent numbers should be represented
+        case similarity(array) == true:
+            console.log(`Invalid! The input ${num} should contain at least 2 different types of numbers!`);
+            break;
+
+        //The last number must be even
+        case array.pop() % 2 == 0:
```

Comment:
when you `pop` you're removing the last item. The array will be one item smaller. Use an index instead